src/storage/src/controller/hosts.rs

```diff
@@ -148,81 +143,66 @@ where
         StorageCommand<T>: RustType<ProtoStorageCommand>,
         StorageResponse<T>: RustType<ProtoStorageResponse>,
     {
-        let (host_addr, orchestrated) = match host_config {
-            StorageHostConfig::Remote { addr } => (addr, false),
+        let host_addr = match host_config {
+            StorageHostConfig::Remote { addr } => {
+                self.drop_storage_host(id).await?;
+                addr
+            }
             StorageHostConfig::Managed { allocation, .. } => {
-                (self.start_storage_host(id, allocation).await?, true)
+                self.ensure_storage_host(id, allocation).await?
```

Comment:
Do we have guarantees that the address returned here will be of the new pod? I don't know for how long `ensure_service` blocks in this case.

(that said, if it returns something while two pods are up, the code will panic, so this will be very noticeable. Also, thinking a bit more about it, it's a `StatefulSet`, so we shouldn't have two pods up at the same time anyway)

Reply:
I think specifically the way this works for stateful sets is that the address won't change, but the pod it addresses will. So normally reconfiguring the service won't actually change the address.

I don't think the orchestrator guarantees that the new pod will be up and listening at that address by the time this returns. But assuming the new pod does come up as expected, it should be listening at that address... same behaviour as before, AIUI.